app/victoria-metrics/main_test.go

```diff
@@ -63,26 +63,34 @@ var (
 )
 
 type test struct {
-	Name   string `json:"name"`
-	Data   string `json:"data"`
-	Query  string `json:"query"`
-	Result []Row  `json:"result"`
-	Issue  string `json:"issue"`
+	Name    string   `json:"name"`
+	Data    string   `json:"data"`
+	Query   string   `json:"query"`
+	Metrics []Metric `json:"metrics"`
+	Series  Series   `json:"series"`
+	Issue   string   `json:"issue"`
 }
 
-type Row struct {
+type Metric struct {
 	Metric     map[string]string `json:"metric"`
 	Values     []float64         `json:"values"`
 	Timestamps []int64           `json:"timestamps"`
 }
 
-func (r *Row) UnmarshalJSON(b []byte) error {
-	type withoutInterface Row
+const seriesStatusSuccess = "success"
+
+type Series struct {
+	Status string              `json:"status"`
+	Data   []map[string]string `json:"data"`
+}
+
+func (r *Metric) UnmarshalJSON(b []byte) error {
+	type withoutInterface Metric
 	var to withoutInterface
 	if err := json.Unmarshal(populateTimeTpl(b), &to); err != nil {
```

Comment:
Why not unmarshaling directly to `r`?

Reply:
because r *Metric has UnmarshalJSON interface which be called by json.Unmarshal and func goes to infinite loop

Reply:
Try 
```
func (r *Metric) UnmarshalJSON(b []byte) error {
	type plain Metric
	return json.Unmarshal(populateTimeTpl(b), (*plain)(r))
}
```